### PR TITLE
LinalgRefactor - rm Apply and ElementwiseProduct

### PR DESCRIPTION
**Pull request by OXPHOS**

### CODE REVIEW

File: src/shogun/lib/GPUMatrix.h [hunk 1]
```diff
@@ -41,6 +41,7 @@
 
 #include <shogun/lib/common.h>
 #include <memory>
+#include <viennacl/linalg/matrix_operations.hpp>
```
Comment (OXPHOS) [inline src/shogun/lib/GPUMatrix.h]: This was in `Apply.h` and it is required for GPU matrix element access.
NOTE: current new GPUMatrix/Vector doesn't support element access. Maybe we can add it back.

Comment (karlnapf) [inline src/shogun/lib/GPUMatrix.h]: Nope, no element access for GPU structures (forces user to transfer back to cpu first)

Comment (karlnapf) [inline src/shogun/lib/GPUMatrix.h]: Nope, no element access for GPU structures (forces user to transfer back to cpu first)

Comment (karlnapf) [inline src/shogun/lib/GPUMatrix.h]: The point is that element-wise product on the GPU is done in vectorised form

**Pull request merged**